src/components/ItemsList.js

```diff
@@ -43,43 +44,60 @@ const ItemsList = () => {
       {loading && <p>Loading...</p>}
       {error && <p>An error has occured...</p>}
-      {shoppingList && !shoppingList.length && (
-            <div className="add-item">
-              <p>You haven't created a shopping list yet...</p>
-              <button type="submit" onClick={handleRedirect}>
-                Add First Item
-              </button>
-            </div>
-          )}
-      <form>
-        <ul>
-          {shoppingList &&
-            shoppingList[0] &&
-            shoppingList[0].items.map((shoppingItemObject, index) => {
-              return (
-                <li key={shoppingItemObject.shoppingListItemName + index}>
-                  <input
-                    type="checkbox"
-                    id={shoppingItemObject.shoppingListItemName}
-                    onChange={() => markItemAsPurchased(index)}
-                    checked={wasItemPurchasedWithinLastOneDay(
-                      shoppingItemObject.lastPurchasedOn,
-                    )}
-                  />
-                  <label htmlFor={shoppingItemObject.shoppingListItemName}>
-                    {shoppingItemObject.shoppingListItemName}
-                  </label>
-                </li>
-              );
-            })}
-        </ul>
-      </form>
+      {listHasNoItems && (
+        <div className="add-item">
+          <p>You haven't created a shopping list yet...</p>
+          <button type="submit" onClick={handleRedirect}>
+            Add First Item
+          </button>
+        </div>
+      )}
+      {listHasAtLeastOneItem ? (
+        <div className="search-list">
+          <SearchBar
+            value={searchTerm}
+            setValue={(searchTerm) => {
+              setSearchTerm(searchTerm);
+            }}
+          />
+          <ul>
+            {shoppingList[0].items
+              .filter((shoppingItemObject) =>
+                shoppingItemObject.shoppingListItemName
+                  .toLowerCase()
+                  .includes(searchTerm.toLowerCase()),
+              )
+              .map((shoppingItemObject, index) => {
+                const shopIndex = shoppingList[0].items.indexOf(
+                  shoppingItemObject,
+                );
+                return (
+                  <li key={shoppingItemObject.shoppingListItemName + index}>
+                    <input
+                      type="checkbox"
+                      id={shoppingItemObject.shoppingListItemName}
+                      onChange={() => markItemAsPurchased(shopIndex)}
+                      checked={wasItemPurchasedWithinLastOneDay(
+                        shoppingItemObject.lastPurchasedOn,
+                      )}
+                    />
+                    <label htmlFor={shoppingItemObject.shoppingListItemName}>
+                      {shoppingItemObject.shoppingListItemName}
+                    </label>
+                  </li>
+                );
+              })}
+          </ul>
+        </div>
+      ) : null}
```

Comment:
For this block you can use `&&`  like you used for line 55 so you don't need to return the null at the end